glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -173,9 +85,33 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
          * @return true if process was successful
          */
         internal fun uploadPings(): Boolean {
-            val storageDirectory = File(Glean.getDataDir(), PINGS_DIR)
+            var ffiTask = LibGleanFFI.INSTANCE.glean_get_upload_task()
+            var task = ffiTask.toPingUploadTask()
+            while (task is PingUploadTask.Upload) {
+                // Get the request
+                val request = task.request()!!
```

Comment:
Wow, you are correct! Weird, I created this function because when I first started messing with this it would complain when I tried `task.request` it says `task` didn't have that property. Oh well, glad I can do it now :D